backend/db/repos/leaderboard_repository.go

```diff
@@ -55,12 +60,35 @@ func (lr *LeaderboardRepository) GetLeaderboard() ([]*LeaderboardEntry, error) {
 			return nil, err
 		}
 
+		// Count number of rows created after the last time the user lost a game
+		lr.db.Model(&models.Team{}).
+			Where("(user_one_id = ? OR user_two_id = ?)", user.ID, user.ID).
+			Where("id > (SELECT MAX(id) from teams WHERE winner = false AND (user_one_id = ? OR user_two_id = ?))", user.ID, user.ID).
+			Count(&teamCounts.WinningStreak)
+
+		if teamCounts.Losing == 0 {
```

Comment:
I had a really hard time wrapping my head around this if. Maybe we could add a comment saying: "if user has never lost, their win streak is amount of wins"

It took me a while to realize it's because when you haven't lost the above will return 0.

Reply:
Added comment